tree/treeplayer/src/TSimpleAnalysis.cxx

```diff
@@ -0,0 +1,252 @@
+      line = line.substr(firstNotSpace, line.size()-firstNotSpace);
+   std::size_t lastNotSpace = line.find_last_not_of(" ");
+   if (lastNotSpace != std::string::npos)
+      line = line.substr(0, lastNotSpace+1);
+   return 1;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// It handles the expression lines of the input file in order to pass the
+/// elements to the members of the object
+///
+/// param[in] line line read from the input file or the expression passed to the constructor
+
+std::string TSimpleAnalysis::HandleExpressionConfig(std::string& line)
+{
+   std::size_t equal = line.find("=");
+   if (equal == std::string::npos)
+      return "Error: missing '='";
+   std::size_t cutPos = line.find(kCutIntr, equal);
+   std::string histName = line.substr(0, equal);
+   if (histName.empty())
+      return "Error: no histName found";
+   DeleteSpaces(histName);
+
+   std::string histExpression = line.substr(equal+1, cutPos-equal-1);
+   if (histExpression.empty())
+      return "Error: no expression found";
+   DeleteSpaces(histExpression);
+
+   std::string histCut;
+   if (cutPos != std::string::npos)
+      histCut = line.substr(cutPos+kCutIntr.size());
+
+   auto check = fHists.insert(std::make_pair((const std::string&)histName,
+                                             std::make_pair(histExpression, histCut)));
+   if (!check.second)
+      return "Name of the histogram already existing";
+   return "";
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Constructor for the case with the input file
+///
+/// \param[in] kFile name of the input file that has to be read
+
+TSimpleAnalysis::TSimpleAnalysis(const std::string& file): fInputName(file)
+{}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Constructor for the case of command line parsing arguments
+///
+/// \param[in] output name of the output file
+/// \param[in] inputFiles name of the input .root files
+/// \param[in] name name of the tree
+/// \param[in] expressions what is showed in the histograms
+
+TSimpleAnalysis::TSimpleAnalysis(const std::string& output, const std::vector<std::string>& inputFiles,
+                                 const std::string& name, std::vector<std::string> expressions):
+   fInputFiles(inputFiles), fOutputFile(output), fTreeName(name)
+{
+      for (std::string expr: expressions) {
+        std::string errMessage = HandleExpressionConfig(expr);
+         if (!errMessage.empty())
+            throw std::runtime_error(errMessage + " in " +  expr);
+      }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Return true if the input arguments create the output file correctly
+
+bool TSimpleAnalysis::Analyze()
+{
+   TChain chain(fTreeName.c_str());
+   for (const std::string& inputfile: fInputFiles)
+      chain.Add(inputfile.c_str());
+   TFile ofile(fOutputFile.c_str(), "RECREATE");
```

Comment:
Before opening ofile, call `chain.LoadTree(0)` to ensure that the chain is correctly set up, and that the first input file is not an expression, e.g. from the ill-formed case
```
input1.root

hpx = x
```